### DISCUSSION TITLE
Caching Respones

### DISCUSSION THREAD

**Discussion by hjaber** [PostgREST/postgrest#3087] [Q&A]
I am using a cloudflare worker to fetch data from supabase via postgREST. I'd like to cache certain queries but not all, however, I noticed supabase's postgREST is using cloudflare and I do not believe you can cache another account's cloudflare worker.

Is it possible to cache responses per query? The [Response Headers](https://postgrest.org/en/stable/references/transactions.html#response-headers) section in the docs makes it seem like it's possible through an sql statement but not necessarily from a fetch() with headers 

supabase postgREST response headers
```
HTTP/2 200 
date: Mon, 04 Dec 2023 06:05:29 GMT
content-type: application/json; charset=utf-8
content-range: 0-7/*
cf-ray: 8301dba17988fac2-SJC
cf-cache-status: DYNAMIC
access-control-allow-origin: *
content-location: /todos?select=%2A
strict-transport-security: max-age=2592000; includeSubDomains
vary: Accept-Encoding
via: kong/2.8.1
content-profile: public
sb-gateway-version: 1
x-kong-proxy-latency: 0
x-kong-upstream-latency: 17
server: cloudflare
alt-svc: h3=":443"; ma=86400
```

**Comment by steve-chavez** [comment]
Not experienced with cloudflare. Can the problem be isolated from that?

> Is it possible to cache responses per query? The [Response Headers](https://postgrest.org/en/stable/references/transactions.html#response-headers) section in the docs makes it seem like it's possible through an sql statement

Using [pre-request](https://postgrest.org/en/stable/references/transactions.html#pre-request) and [request.path](https://postgrest.org/en/stable/references/transactions.html#request-role-and-search-path) it should be possible to set different headers per query. So a request on `/table` would have different headers than a request on `/rpc/func`.

> but not necessarily from a fetch() with headers

Not sure I understand the above but a client can send a header, you can get this header with SQL and do some conditional logic.

**Reply by hjaber** [reply]
Thanks @steve-chavez, I suspect you are correct and this is more related to Supabase's integration with Cloudflare

>but not necessarily from a fetch() with headers

I think I was looking for a more straightforward way to set response headers rather than creating an rpc so that something like below could be configured on the fly
```ts
fetch("https://postgrest-server.com/todos?select=*", {
  headers: {
    "Cache-Control": "max-age=60",
  },
});
```

**Comment by wolfgangwalther** [comment]
Closing as a supabase related question, not PostgREST.
